artemis-server/src/main/java/org/apache/activemq/artemis/core/paging/impl/PagingStoreImpl.java

```diff
@@ -343,6 +344,8 @@ private void checkNumberOfPages() {
       if (!isBelowPageLimitBytes()) {
          this.pageFull = true;
          ActiveMQServerLogger.LOGGER.pageFullMaxBytes(storeName, numberOfPages, estimatedMaxPages, pageLimitBytes, pageSize);
+      } else {
+         pageLimitReleased();
```

Comment:
This seems wrong. Just because it isnt over the file-based paging limit, doesnt mean it isnt still 'full'; there is a separate message-based limit which this would appear to essentially disregard, effectively bypassing it any time this method is called...see related use of `checkPageLimit(long numberOfMessages)` (next method right below this).